compiler/src/dotty/tools/dotc/core/TypeOps.scala

```diff
@@ -512,7 +512,7 @@ object TypeOps:
       @threadUnsafe lazy val forbidden = symsToAvoid.toSet
       def toAvoid(tp: NamedType) =
         val sym = tp.symbol
-        !sym.isStatic && forbidden.contains(sym)
```

Comment:
But shouldn't the anonymous subclass be considered non-static since it's declared in a block? I assume the issue is that its owner is set to the top-level scope, but if we can't change its owner we could perhaps special case ANON_CLASS in SymDenotation#isStatic